src/types/job.rs

```diff
@@ -41,6 +41,22 @@ pub struct JobDescriptor {
     pub num_incomplete: u32,
 }
 
+/// Either a list of package descriptors or a list of PURLs.
+#[derive(
+    Clone, Debug, Deserialize, Eq, Hash, JsonSchema, Ord, PartialEq, PartialOrd, Serialize,
+)]
+#[serde(untagged)]
+pub enum PackageDescriptorsOrPurls {
+    PackageDescriptors(Vec<PackageDescriptor>),
+    Purls(Vec<String>),
```

Comment:
Including the `Vec<_>` in each variant seems like a bad idea. The [single package submit](https://api.staging.phylum.io/api/v0/swagger/index.html#/Packages/submit_endpoint) endpoint only accepts a single package, so it won't be able to use this type.

I am guessing this was done to force users to submit either all PURLs or all descriptor objects and not mix the two. But if the API supports both, why not let them be mixed?